docker/archive/dest.go

```diff
@@ -19,17 +19,32 @@ func newImageDestination(ctx *types.SystemContext, ref archiveReference) (types.
 	if ref.destinationRef == nil {
 		return nil, errors.Errorf("docker-archive: destination reference not supplied (must be of form <path>:<reference:tag>)")
 	}
-	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+	// tries to open a file first to accomodate for load when using stdout
+	// as the file is created first then written to
+	// if the file doesn't exist then it creates a new file for exclusive use
+	// if the file exists, it checks to ensure the size is 0
+	fh, err := os.OpenFile(ref.path, os.O_WRONLY, 0644)
 	if err != nil {
-		// FIXME: It should be possible to modify archives, but the only really
-		//        sane way of doing it is to create a copy of the image, modify
-		//        it and then do a rename(2).
-		if os.IsExist(err) {
-			err = errors.New("docker-archive doesn't support modifying existing images")
+		fh, err = os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
```

Comment:
I don’t think using `O_EXCL` with the same path again is all that valuable; either `err` was an unrelated failure and this should fail again in the same way, or we already know that a file did not exist, so if it exists now and `O_EXCL` causes us to fail, we are racing against something else modifying the filesystem and failing is not necessarily the right thing to do.

Couldn’t this be simplified to a single `os.OpenFile(…O_WRONLY | O_CREATE)` call followed by the “size of the file (whether pre-existing or newly created) is 0” check? (That’s also racy against concurrent writes to the same filename, though differently, but at least it is trivially simple.)